File: CustomizablePasswordGenerator.cs
using System;
using System.Text;

namespace CustomizablePasswordGenerator
{
    
    class CustomizablePasswordGenerator
    {
        public static void ValidateInput(int minLength, int maxLength, int numberOfDigits, int numberOfSpecialChars)
        {
            if (maxLength < minLength)
                throw new Exception("Minimum length cannot be bigger than max length!");

            if (numberOfDigits + numberOfSpecialChars > maxLength)
                throw new Exception("Sum of digit and symbol numbers cannot be bigger than max length");

        }

        public static String GeneratePassword(int minLength, int maxLength, int numberOfDigits, int numberOfSpecialChars)
        {
            char[] numbers = "0123456789".ToCharArray();
            char[] symbols = "^$?!@#%&".ToCharArray();
            char[] allAllowed = "abcdefghijklmnopqrstuvwxyzABCDEFGJKLMNPRSTUVWXYZ0123456789^$?!@#%&".ToCharArray();

            Random random = new Random();
            StringBuilder password = new StringBuilder();
            for (int i = 0; i < numberOfDigits; i++)
            {
                password.Insert(random.Next(password.Length + 1),numbers[random.Next(numbers.Length)]);
            }
            for (int i = 0; i < numberOfSpecialChars; i++)
            {
                password.Insert(random.Next(password.Length + 1),symbols[random.Next(symbols.Length)]);
            }
            int randomLength = maxLength;
            if (maxLength - minLength > 0)
            {
                randomLength = random.Next(maxLength - minLength + 1) + minLength;
            }
            int remainingLength = randomLength - numberOfDigits - numberOfSpecialChars;
            for (int i = 0; i < remainingLength; i++)
            {
                password.Insert(random.Next(password.Length + 1),allAllowed[random.Next(allAllowed.Length)]);
            }

            return password.ToString();

        }

    }
}

File: CustomizablePasswordGenerator.java
import java.security.SecureRandom;
import java.util.Random;
import java.util.Scanner;

public class CustomizablePasswordGenerator {

    private static void validateInput(int min_len, int max_len, int numberOfDigits, int numberOfSpecialChars) {
        if (max_len < min_len)
            throw  new IllegalArgumentException("Minimum length cannot be bigger than max length!");

        if (numberOfDigits+ numberOfSpecialChars > max_len)
            throw new IllegalArgumentException("Sum of digit and symbol numbers cannot be bigger than max length");

    }

    public static String generatePassword(int min_len, int max_len ,int numberOfDigits, int numberOfSpecialChars) {

        final char[] lowercase = "abcdefghijklmnopqrstuvwxyz".toCharArray();
        final char[] uppercase = "ABCDEFGJKLMNPRSTUVWXYZ".toCharArray();
        final char[] numbers = "0123456789".toCharArray();
        final char[] symbols = "^$?!@#%&".toCharArray();
        final char[] allAllowed = "abcdefghijklmnopqrstuvwxyzABCDEFGJKLMNPRSTUVWXYZ0123456789^$?!@#%&".toCharArray();

        Random random = new SecureRandom();
        StringBuilder password = new StringBuilder();
        for (int i = 0; i < numberOfDigits; i++) {
            password.insert(random.nextInt(password.length()+1), numbers[random.nextInt(numbers.length)]);
        }
        for (int i = 0; i < numberOfSpecialChars; i++) {
            password.insert(random.nextInt(password.length()+1), symbols[random.nextInt(symbols.length)]);
        }
       int randomLength = max_len;
       if (max_len- min_len > 0){
           randomLength = random.nextInt(max_len-min_len+1) +min_len;
       }
       int remainingLength = randomLength - numberOfDigits - numberOfSpecialChars;
        for (int i = 0; i < remainingLength; i++) {
            password.insert(random.nextInt(password.length()+1), allAllowed[random.nextInt(allAllowed.length)]);
        }

        return password.toString();

    }

}



